client-core/src/wallet/syncer.rs

```diff
@@ -277,14 +307,74 @@ impl<'a, S: SecureStorage, C: Client, D: TxDecryptor, F: FnMut(ProgressReport) -
         Ok(())
     }
 
+    pub fn handle_recover_addresses_for_transaction(
+        &mut self,
+        transaction: &Transaction,
+    ) -> Result<()> {
+        let mut refetch = false;
+
+        let outputs = transaction.outputs().to_vec();
+
+        for (_i, output) in outputs.iter().enumerate() {
+            let newaddress = output.address.to_string();
+            let tmp_refetch = self.env.recover_address.recover_addresses(
+                &newaddress,
```

Comment:
`output.address` could be passed directly?

Reply:
yes,i'll change it